scipy/special/_basic.py

```diff
@@ -2209,6 +2215,12 @@ def comb(N, k, exact=False, repetition=False):
     if repetition:
         return comb(N + k - 1, k, exact)
     if exact:
+        if int(N) != N or int(k) != k:
+            warnings.warn(
+                "Non-integer arguments are deprecated when exact=True; "
+                "note that arguments will be cast to integers (as before)",
```

Comment:
I think it'd be nice to say when this will be completely removed. Some recent deprecations (gh-15512, gh-15573) just say "... will be removed in the second release after SciPy 1.9.0" since version number after 1.9.0 is TBD (gh-15528):

```suggestion
                "Non-integer arguments are deprecated when exact=True "
                "and will be removed in the second release after "
                "SciPy 1.9.0",
```